pyproject.toml

```diff
@@ -48,5 +48,10 @@ addopts = "-ra"
 
 [tool.ruff]
 target-version = "py39"
-select = ["B", "E", "F", "I", "N", "W", "PL", "PT", "UP", "RUF"]
-ignore = ["PLR2004"]
+lint.select = ["B", "E", "D", "F", "I", "N", "S", "W", "PL", "PT", "UP", "RUF"]
+# D104: Missing docstring in public package
+# D203: 1 blank line required before class docstring
+# D213: Multi-line docstring summary should start at the second line
+# S101: Use of assert detected
```

Comment:
Should this only be in the tests directly under a `per-file-ignore` regex? I don't think we want assert, False, "message" in the main code and would prefer raising an exception instead.

Reply:
Good point, I do think that is the better way to go on this one. I will update that.